misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/PerformanceAnalyzerService.java

```diff
@@ -74,7 +79,27 @@ public void init(LanguageServer langServer, WorkspaceManager workspaceManager) {
 
         return CompletableFuture.supplyAsync(() -> {
             String fileUri = request.getDocumentIdentifier().getUri();
-            return EndpointsFinder.getEndpoints(fileUri, this.workspaceManager, request.getRange());
+            JsonObject data = EndpointsFinder.getEndpoints(fileUri, this.workspaceManager, request.getRange());
+
+            if (data.entrySet().isEmpty()) {
+                JsonObject obj = new JsonObject();
+                obj.addProperty(TYPE, ERROR);
+                obj.addProperty(MESSAGE, ENDPOINT_RESOLVE_ERROR);
```

Comment:
Shouldn't this be a meaningful message? We are adding a message code.

Reply:
Error messages are added in the front-end side based on these error codes